From 39090d39583a80ed4d412bed41e837bed8414afa Mon Sep 17 00:00:00 2001
From: Maciej Blizinski <maciej@opencsw.org>
Date: Wed, 18 Sep 2013 09:43:43 +0200
Subject: [PATCH] include/pcap must be first to pick pcap_file()

If /opt/csw/include/pcap.h gets picked up first, it and only it gets parsed
and searched for "pcap_file(". But /opt/csw/include/pcap.h is only
an empty compatibility file, it only includes the /opt/csw/include/pcap/pcap.h
file. But since this code only parses the first found include file,
.../include/pcap/pcap.h doesn't get parsed and the configure script thinks
that the pcap_file() function is not available.

The downsides of custom configure scripts with custom discovery methods, not
actually running the compiler.

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 24eaa1e..7a0728f 100644
--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,7 @@ class config_pcap(config.config):
             dirs = [ '/usr', sys.prefix ] + glob.glob('/opt/libpcap*') + \
                    glob.glob('../libpcap*') + glob.glob('../wpdpack*')
         for d in dirs:
-            for sd in ('include', 'include/pcap', ''):
+            for sd in ('include/pcap', 'include', ''):
                 incdirs = [ os.path.join(d, sd) ]
                 if os.path.exists(os.path.join(d, sd, 'pcap.h')):
                     cfg['include_dirs'] = [ os.path.join(d, sd) ]
-- 
1.8.3.4

